### DISCUSSION TITLE
Can't delete files in GitHub web GUI

### DISCUSSION THREAD

**Discussion by malonzm1** [community/community#67785] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi,

I can't seem to delete files in the GitHub web GUI. There's no trash bin icon nor a delete option in the dropdown menu. 
Please advise.

<img width="1258" alt="Screenshot 2023-09-21 at 1 13 05 PM" src="https://github.com/community/community/assets/32474661/995b9da3-d9a9-4bf3-8ffe-022467a60bae">

Thanks and good day.

**Comment by KiranZafar23** [comment]
1. **Navigate to the Repository**: Go to the GitHub repository where the file is located.

2. **Browse to the File**: Locate the file you want to delete within the repository.

3. **Edit the File**: Click on the file's name to view its content.

4. **Edit Button**: At the top right of the file content, you'll see an "Edit" button. Click on it.

5. **Delete the Content**: Delete the entire content of the file. This effectively empties the file.

6. **Commit Changes**: Scroll down, and you'll find a section to "Commit changes." Here, you can provide a commit message explaining the deletion.

7. **Commit Changes Button**: Click the "Commit changes" button to confirm the deletion.

This process effectively deletes the file from the repository. If you want to delete multiple files or an entire folder, you can follow the same process for each file within the folder.

Remember that this will create a commit in your repository's history, indicating the deletion of the file. Make sure you have the necessary permissions to make changes to the repository.

If you want to restore a deleted file or view its history, you can do so through the repository's commit history.

This method uses the GitHub web interface's built-in file editing capabilities to delete files. If you prefer a command-line approach or need more advanced file management, you might consider using Git commands locally and pushing the changes to the repository.

**Comment by malonzm1** [comment]
Thanks. I followed your instructions. Though I've deleted the entire content of the file, the filename is still visible in the folder. The file is still there, it's just empty now. Please advise.

**Reply by KiranZafar23** [reply]
To delete a file on GitHub:

1. Edit the file and delete its content.
2. Below the editor, find "Commit changes."
3. Provide a commit message (e.g., "Delete [filename]").
4. Click "Commit changes."

This will remove the file from the repository.

**Reply by malonzm1** [reply]
Thanks. I have followed these four steps (same as the seven stops above). The file is still in the repository, it's just blank.

**Reply by ttelang** [reply]
**For everyone’s convenience:**
You most likely won’t need to reach out to your admin or GitHub support.
Just scroll down in the top-right menu, and you’ll find the Delete option at the bottom.

🔍 Check the screenshots here:
https://github.com/orgs/community/discussions/67785#discussioncomment-9022809

**Comment by greatlarrymartin** [comment]
If you are unable to delete files in the GitHub web GUI and you are unable to find a trash bin icon or a delete option in the dropdown menu, there are a few possible solutions you can try:

Make sure you have the necessary permissions: Check if you have the necessary access rights to delete files in the repository. If you do not have the required permissions, reach out to the repository owner or administrator and request the appropriate access.
Try a different browser: Sometimes, certain browsers may not display all the features correctly. Switching to a different browser or updating your current browser to the latest version might help resolve the issue.
Use the command line: If the web GUI is not providing the required delete functionality, you can try deleting the files using the command line. You can clone the repository to your local machine, navigate to the desired file, and delete it using Git commands. Don't forget to push the changes afterward.
If none of these solutions work, consider reaching out to the GitHub support team or posting your issue on their community forum for further assistance.

**Reply by malonzm1** [reply]
Thanks.

**Reply by tiassap** [reply]
@malonzm1 

I had the same issue when using Chrome browser. But if you scroll down on the dropdown button (three dots icon), you will find the delete file button. On Safari, the delete button is visible but partly hidden.

I hope GitHub support team would fix this.

**Reply by sbwalker** [reply, marked as answer]
To explain the issue more clearly... the "Delete file" option is at the end of the drop down list and is not immediately visible in most browsers. You need to notice that there is a vertical scroll bar and then scroll down to reveal the "Delete file" option. It would be useful if the dropdown menu height was set large enough so that the Delete file option was visible to avoid this UX issue.

![image](https://github.com/community/community/assets/4840590/2b8d0c20-dac1-4524-9c84-33f5bdecdf5c)

![image](https://github.com/community/community/assets/4840590/f52909d8-fa68-4c2a-af6c-b4d22c872a25)

**Reply by natancelidonio** [reply]
> To explain the issue more clearly... the "Delete file" option is at the end of the drop down list and is not immediately visible in most browsers. You need to notice that there is a vertical scroll bar and then scroll down to reveal the "Delete file" option. It would be useful if the dropdown menu height was set large enough so that the Delete file option was visible to avoid this UX issue.
> 
> ![image](https://private-user-images.githubusercontent.com/4840590/320046822-2b8d0c20-dac1-4524-9c84-33f5bdecdf5c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxODg3OTAsIm5iZiI6MTcxNzE4ODQ5MCwicGF0aCI6Ii80ODQwNTkwLzMyMDA0NjgyMi0yYjhkMGMyMC1kYWMxLTQ1MjQtOWM4NC0zM2Y1YmRlY2RmNWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MzFUMjA0ODEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGJjNGZlYjhhZmY4MTdkNTcyZDM2NTdhYTJhMzk5OGE4OWFjNTFmOTY3MTA0ZTI2ODQwODBiNWMxOTU2YzE2NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1F040H1vGc1ZHdj6vwS2fTBXMDfiGOlpYTw-A4N_3NU)
> 
> ![image](https://private-user-images.githubusercontent.com/4840590/320047052-f52909d8-fa68-4c2a-af6c-b4d22c872a25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxODg3OTAsIm5iZiI6MTcxNzE4ODQ5MCwicGF0aCI6Ii80ODQwNTkwLzMyMDA0NzA1Mi1mNTI5MDlkOC1mYTY4LTRjMmEtYWY2Yy1iNGQyMmM4NzJhMjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MzFUMjA0ODEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTc4MjEyYTlhZTQ0MzAyZDdkZDY3YzY3MDUwNTQ4Mzg0MDA5MmY3MjVjZmYxMmU1YmYwNmJlYWM2YmNhNjRkNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Qo1b4x-4JRF2EBPodNt5UPhzUXFkQ6si71f0Q0oQIDo)

GOD, THANK YOU!!!

**Reply by gefirmans** [reply]
my god! ok i got it, thank you @sbwalker

**Comment by zinoubm** [comment]
I just resized down my browser zoom, then I was able to see the delete option from the drop down.

**Comment by Coolbuster10** [comment]
For me, on windows 11, it seems to be a browser issue. On Google chrome there's no delete option although on Microsoft edge there was. Must be a small bug.

**Comment by ipkpjersi** [comment]
I just ran into this issue. Make sure you are on the correct branch, I was viewing the tree at a specific commit.
